File: glean-core/android/src/main/java/mozilla/telemetry/glean/private/TimeUnit.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.telemetry.glean.private

/**
 * Enumeration of different resolutions supported by
 * the Timespan and DateTime metric types.
 */
enum class TimeUnit {
    /**
     * Represents nanosecond precision.
     */
    Nanosecond,
    /**
     * Represents microsecond precision.
     */
    Microsecond,
    /**
     * Represents millisecond precision.
     */
    Millisecond,
    /**
     * Represents second precision.
     */
    Second,
    /**
     * Represents minute precision.
     */
    Minute,
    /**
     * Represents hour precision.
     */
    Hour,
    /**
     * Represents day precision.
     */
    Day,
}

File: glean-core/ios/Glean/Metrics/TimeUnit.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import Foundation

/// Enumeration of different resolutions supported by
/// the `TimespanMetricType` and `TimingDistributionMetricType`.
public enum TimeUnit: Int32 {
    /// Represents nanosecond precision.
    case nanosecond = 0

    /// Represents microsecond precision.
    case microsecond = 1

    /// Represents millisecond precision.
    case millisecond = 2

    /// Represents second precision.
    case second = 3

    /// Represents minute precision.
    case minute = 4

    /// Represents hour precision.
    case hour = 5

    /// Represents day precision.
    case day = 6
}
